libs/vkd3d/device.c

```diff
@@ -5106,17 +5121,109 @@ static HRESULT STDMETHODCALLTYPE d3d12_device_OpenSharedHandleByName(d3d12_devic
 static HRESULT STDMETHODCALLTYPE d3d12_device_MakeResident(d3d12_device_iface *iface,
         UINT object_count, ID3D12Pageable * const *objects)
 {
-    FIXME_ONCE("iface %p, object_count %u, objects %p stub!\n",
-            iface, object_count, objects);
+    struct d3d12_device *device = impl_from_ID3D12Device(iface);
+    const struct vkd3d_vk_device_procs *vk_procs = &device->vk_procs;
+
+    if (device->vk_info.EXT_pageable_device_local_memory)
+    {
+        uint32_t i;
+        
+        for (i = 0; i < object_count; i++)
+        {
+            ID3D12Heap *heap_iface;
+            ID3D12Resource *resource_iface;
+            ID3D12DeviceChild *object = (ID3D12DeviceChild*)objects[i];
+            VkDeviceMemory memory = VK_NULL_HANDLE;
+            D3D12_RESIDENCY_PRIORITY priority;
+
+            if (SUCCEEDED(ID3D12DeviceChild_QueryInterface(object, &IID_ID3D12Heap, (void**)&heap_iface)))
+            {
+                struct d3d12_heap *heap_object = impl_from_ID3D12Heap(heap_iface);
+
+                memory = heap_object->allocation.device_allocation.vk_memory;
+                priority = heap_object->priority;
```

Comment:
Same concern here, a very small heap without DENY_BUFFERS that is smaller than 2MB will be suballocated and should be exempt from memprio. In practice, it shouldn't happen however.

Reply:
FWIW I think this is happening quite a lot (Last Of Us & Spider-Man, but approximately never in UE4 titles).  Not a problem as such, just wanted to check expectations.